CMakeLists.txt

```diff
@@ -15,36 +17,80 @@ execute_process(
   ERROR_QUIET)
 
 if(ECODE)
-  message(
-    "Failed to determine version from git, using ${CMAKE_PROJECT_VERSION}")
+  message("Failed to determine version from git, using ${CMAKE_PROJECT_VERSION}")
   set(NEURONPARQUET_VERSION ${CMAKE_PROJECT_VERSION})
 endif()
 
+# Find required packages
 find_package(MPI REQUIRED)
 find_package(Arrow REQUIRED)
+
+# Add MPI include directories and definitions
+include_directories(SYSTEM ${MPI_INCLUDE_PATH})
+add_definitions(${MPI_CXX_COMPILE_FLAGS})
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${MPI_CXX_LINK_FLAGS}")
+add_definitions(-DOMPI_SKIP_MPICXX)
 get_filename_component(MY_SEARCH_DIR ${Arrow_CONFIG} DIRECTORY)
 find_package(Parquet REQUIRED HINTS ${MY_SEARCH_DIR})
 find_package(HDF5 REQUIRED)
 find_package(nlohmann_json REQUIRED)
 find_package(Range-v3 REQUIRED)
-
 find_package(HighFive)
+find_package(CLI11)
+find_package(Catch2)
+find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
+
+# Add HighFive if not found
```

Comment:
```suggestion
```
If a comment just says what is written in plain code below, no value is added. Please remove these comments.